.github/workflows/publish.yml

```diff
@@ -0,0 +1,94 @@
+# Publishes built jars to distribution platforms
+name: Publish
+
+on:
+  push:
+    tags:
+      - 'v*.*.*' # any semver tag, e.g. v1.2.3
+
+# if a second commit is pushed quickly after the first, cancel the first one's build
+concurrency:
+  group: publish-${{github.head_ref}}
+  cancel-in-progress: true
+
+env:
+  # link to the changelog with a format code for the version
+  CHANGELOG_LOCATION: "Changelog is available [here](https://github.com/${{github.repository}}/releases/tag/${{github.ref_name}})"
+  # type of release
+  RELEASE_TYPE: "beta"
+  # mark the environment as CI/CD for enabling required tasks
+  CI_CD_ENVIRONMENT: true
```

Comment:
There is `CI` variable which is set to `true` always and is commonly used for this kind of job